src/index.js

```diff
@@ -6,17 +6,26 @@ client.collectDefaultMetrics();
 
 module.exports = {
     client,
-    middleware: async (ctx, next) => {
-        ctx.state.prometheus = client;
-        if (ctx.path === '/metrics') {
-            debug('GET /metrics');
-            if (!ctx.headers["x-forwarded-for"]) {
-                ctx.set('Content-Type', client.register.contentType);
-                ctx.body = client.register.metrics();
-            } else {
-                ctx.throw(403, "Forbidden");
+    middleware: (options = {}) => {
+        const path = options.path || '/metrics';
+        const [headerBlacklist] = [options.headerBlacklist];
+        return async (ctx, next) => {
+            ctx.state.prometheus = client;
+            if (ctx.path === path) {
+                debug('GET /%s', path);
+                if (
+                    !headerBlacklist ||
+                    headerBlacklist.filter(h => {
+                        return Object.keys(ctx.headers).includes(h);
+                    }).length === 0
+                ) {
+                    ctx.set('Content-Type', client.register.contentType);
+                    ctx.body = client.register.metrics();
+                } else {
+                    ctx.throw(403, 'Forbidden');
+                }
             }
-        }
-        await next();
+            await next();
```

Comment:
Not sure you want to pass on to the next middleware if L23 happens, after setting the status/body I think it just return.